### PR TITLE
Don't build samples in test runs

### PR DESCRIPTION
**Pull request by pakrym**
Should fix nightly build failures: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=91270&view=logs&j=3a0721f7-08de-5a3a-a8b2-635b007094aa

### CODE REVIEW

**Review by chidozieononiwu — APPROVED** [review]

**Review by danieljurek — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/ChannelReaderExtensionsTests.cs [hunk 1]
```diff
@@ -198,6 +198,7 @@ await foreach (var item in subscription)
         /// </summary>
         ///
         [Test]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/7330")]
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/ChannelReaderExtensionsTests.cs]: Please don't.   Fix is will be merged as soon as the CI validations pass.

**Review by jsquire — APPROVED** [review]
Pending comment, if you'd be so kind.

**Comment by weshaggard** [comment]
What is the expectation for when samples should get built an executed?

**Comment by pakrym** [comment]
> What is the expectation for when samples should get built an executed?

Built as part of the build. This one is a web app so executed manually.

**Comment by pakrym** [comment]
But you are right, some samples are tests that are expected to run. Let me think about it.

**Comment by jsquire** [comment]
> But you are right, some samples are tests that are expected to run. Let me think about it.

The track two Event Hubs samples are smoke tested as part of the Live tests in nightly runs.  Since you're not seeing build/test failures, I don't think they're being excluded by your filter, which I want to be sure doesn't change.

File: eng/service.proj [hunk 2]
```diff
@@ -11,6 +11,8 @@
     <ExcludeMgmtLib Include="..\sdk\*\*.Management.*\**\*.csproj;..\sdk\*mgmt*\**\*.csproj;" />
     <TestProjects Include="..\sdk\$(ServiceDirectory)\**\tests\**\*.csproj" />
     <SamplesProjects Include="..\sdk\$(ServiceDirectory)\**\samples\**\*.csproj" />
+    <!-- Exclude test projects that don't support net461 -->
+    <SamplesProjects Condition="'$(TargetFramework)' == 'net461'" Remove="..\sdk\core\Microsoft.Extensions.Azure\samples\Microsoft.Extensions.Azure.Samples.csproj" />
```
Comment (weshaggard) [inline eng/service.proj]: I generally only like this approach when it is temporary and has an issue filed to fix. If this is never intended to work perhaps we should come up with an alternative strategy. Is there a reason this project cannot support net461?

Comment (pakrym) [reply eng/service.proj]: Yes, it's aspnet core app, ASP.NET Core doesn't support net461

**Comment by pakrym** [comment]
Talked to @weshaggard offline, we decided to go with disabling sample runs as part of the test. Samples shouldn't be serving as a source of coverage.

**Pull request merged**